api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java

```diff
@@ -415,4 +425,75 @@ public void deleteWorkspace_failedProjectDeletion() {
+
+    workspaceService.updateWorkspaceBillingAccount(workspace, newBillingAccount);
+
+    verify(mockFireCloudService)
+        .updateBillingAccount(workspace.getWorkspaceNamespace(), newBillingAccount);
+    verify(mockFireCloudService, never()).updateBillingAccountAsService(anyString(), anyString());
+    assertThat(workspace.getBillingAccountName()).isEqualTo(newBillingAccount);
+  }
+
+  @Test
+  public void updateBillingAccount_userOwnedToFreeTier() throws Exception {
+    workbenchConfig.featureFlags.enableBillingUpgrade = true;
+    String newBillingAccount = "billing-123";
+    DbWorkspace workspace = dbWorkspaces.get(1); // arbitrary choice of those defined for testing
+    workspace.setBillingAccountName(newBillingAccount);
+    assertThat(workspace.getBillingAccountName()).isEqualTo(newBillingAccount);
+
+    workspaceService.updateWorkspaceBillingAccount(
+        workspace, workbenchConfig.billing.freeTierBillingAccountName());
+
+    verify(mockFireCloudService)
+        .updateBillingAccountAsService(
+            workspace.getWorkspaceNamespace(),
+            workbenchConfig.billing.freeTierBillingAccountName());
+    verify(mockFireCloudService, never()).updateBillingAccount(anyString(), anyString());
+    assertThat(workspace.getBillingAccountName())
+        .isEqualTo(workbenchConfig.billing.freeTierBillingAccountName());
+  }
+
+  @Test
+  public void updateBillingAccount_noChange() throws Exception {
+    workbenchConfig.featureFlags.enableBillingUpgrade = true;
+    String newBillingAccount = "billing-123";
+    DbWorkspace workspace = dbWorkspaces.get(1); // arbitrary choice of those defined for testing
+    workspace.setBillingAccountName(newBillingAccount);
+
+    workspaceService.updateWorkspaceBillingAccount(workspace, newBillingAccount);
+
+    verify(mockFireCloudService, never()).updateBillingAccountAsService(anyString(), anyString());
+    verify(mockFireCloudService, never()).updateBillingAccount(anyString(), anyString());
+  }
+
+  @Test
+  public void updateBillingAccount_accountNotOpen() throws Exception {
+    workbenchConfig.featureFlags.enableBillingUpgrade = true;
+    String newBillingAccount = "billing-123";
+    DbWorkspace workspace = dbWorkspaces.get(1); // arbitrary choice of those defined for testing
+    workspace.setBillingAccountName(workbenchConfig.billing.freeTierBillingAccountName());
+    BillingAccount billingAccount = new BillingAccount();
+    billingAccount.setOpen(false);
+    when(mockCloudBillingClient.getBillingAccount(newBillingAccount)).thenReturn(billingAccount);
+
+    assertThrows(
+        BadRequestException.class,
+        () -> workspaceService.updateWorkspaceBillingAccount(workspace, newBillingAccount));
+    verify(mockFireCloudService, never()).updateBillingAccountAsService(anyString(), anyString());
+    verify(mockFireCloudService, never()).updateBillingAccount(anyString(), anyString());
+    verifyZeroInteractions();
```

Comment:
hmm... is this redundant to the previous two lines?

Reply:
Yes!